model/jpa/src/main/java/org/keycloak/connections/jpa/util/JpaUtils.java

```diff
@@ -56,9 +58,10 @@ public class JpaUtils {
     private static final Logger logger = Logger.getLogger(JpaUtils.class);
 
     public static String getTableNameForNativeQuery(String tableName, EntityManager em) {
-        String schema = (String) em.getEntityManagerFactory().getProperties().get(HIBERNATE_DEFAULT_SCHEMA);
         final Dialect dialect = em.getEntityManagerFactory().unwrap(SessionFactoryImpl.class).getJdbcServices().getDialect();
-        return (schema==null) ? tableName : dialect.openQuote() + schema + dialect.closeQuote() + "." + tableName;
+        IdentifierHelper identifierHelper = em.getEntityManagerFactory().unwrap(SessionFactoryImpl.class).getJdbcServices().getJdbcEnvironment().getIdentifierHelper();
+        String schema = em.getEntityManagerFactory().unwrap(SessionFactoryImpl.class).getSessionFactoryOptions().getDefaultSchema();
+        return (schema==null) ? tableName : identifierHelper.toIdentifier(schema).render(dialect) + "." + tableName;
```

Comment:
The initial suggestion used the Liquibase setting, but I think this is mixing JPA and Liquibase items up more than they should be mixed up. And when quoting the schema, we should apply the same logic as Hibernate applies - only quote it when there are special characters. 

I suggest to keep the rest of the code untouched. I added a separate commit, but then I found that there were some conflicts in the intermediate commit, so I squashed your and my changes. 

Please have a look and let me know.

Reply:
Thanks for pushing the change. I agree that your approach is cleaner and it makes sense to utilise JPA/Hibernate here. I wasn't aware of `IdentifierHelper`, good to know :+1: